### PR TITLE
change rendering of list element in sidebar

### LINKED ISSUE #9195: Tweaks to Lists in Sidebar
For the Lists element in the Sidebar:

If the user has 0 lists, remove the Lists element.

tbt11

If the user has 1 or more lists, retain the Lists element, but remove the ADD and MANAGE links.

tbt2

This can happen anytime before the refresh launch.

### PR DESCRIPTION
**Pull request by jchimienti89**
if list count is 0, return null for lists element.

remove add and manage links.

Fixes #9195.

### CODE REVIEW

**Comment by jchimienti89** [comment]
With no list count:

after-styling-sidebar

With list count.

Since I didn't have any lists I tested the component by putting 2 as a the property value of count.
after-styling-sidebar-add-list

**Comment by blowery** [comment]
Thanks again for this work! There's just one problem with it; it removes both the Tags list and the Lists list from the sidebar when there are no current elements, and it removes the Add button from Tags header. We're planning on keeping Tags working the way they are now.

You could probably address this with a new prop to the ExpandableSidebarAddForm, `hideAddButton`, and by moving the suppression of the Lists element out of `ExpandableSidebarMenu` and into the spot in `Sidebar` where we render the Lists menu.

Would you mind making those changes? Thanks again for all the work!

**Comment by jchimienti89** [comment]
Thanks for the clear directions @blowery 

> We're planning on keeping Tags working the way they are now.

Yes, I misunderstood @fraying  just wanted to edit the lists, while leaving the Tags alone.

> moving the suppression of the Lists element out of ExpandableSidebarMenu and into the spot in Sidebar where we render the Lists menu.

Much success! 

```javascript
{ this.props.subscribedLists.length
				? <ReaderSidebarLists
						lists={ this.props.subscribedLists }
						path={ this.props.path }
						isOpen={ this.props.isListsOpen }
						onClick={ this.props.toggleListsVisibility }
						currentListOwner={ this.state.currentListOwner }
						currentListSlug={ this.state.currentListSlug }
					/>
				: null
				}
```

On hiding only lists hide button

> address this with a new prop to the ExpandableSidebarAddForm, hideAddButton

my first attempt was simply to do 

```javascript
		<ExpandableSidebarAddForm hideAddButton={ ! this.props.count } addLabel={ this.props.addLabel } addPlaceholder={ this.props.addPlaceholder } onAddClick={ this.props.onAddClick } onAddSubmit={ this.props.onAddSubmit } />
```

however, this also hid the button for the tags.

my second approach was:

`const shouldHideAddButton = ! this.props.count && this.props.title === 'Lists';`

```javascript
<ExpandableSidebarAddForm hideAddButton={ shouldHideAddButton } addLabel={ this.props.addLabel } addPlaceholder={ this.props.addPlaceholder } onAddClick={ this.props.onAddClick } onAddSubmit={ this.props.onAddSubmit } />
```
but I'm not sure if this explicit call will fly w/ you guys.

**Comment by jchimienti89** [comment]
after-changing-list-component

File: client/reader/sidebar/expandable.jsx [hunk 1]
```diff
@@ -37,11 +37,11 @@ const ExpandableSidebarMenu = React.createClass( {
 				'is-togglable': true
 			}
 		);
-
+		const shouldHideAddButton = ! this.props.count && this.props.title === 'Lists';
```
Comment (blowery) [inline client/reader/sidebar/expandable.jsx]: I'd lift this out to `reader/sidebar/reader-sidebar-lists/index` and pass it through to `Expandable` via a prop. And then just default the prop to true.

**Comment by blowery** [comment]
Left a comment on the new code. That works, but i think it would be better to lift it out to the list sidebar component and pass it in that way. That way you don't really need the conditional on the title name.

**Comment by jchimienti89** [comment]
>  That works, but i think it would be better to lift it out to the list sidebar component and pass it in that way. That way you don't really need the conditional on the title name.

Awesome even simpler. I knew there should be a way to avoid including the title name.

> And then just default the prop to true.

I set the default Prop to false.
Setting it to true was removing the add button from the Tags 

how I tested:
Added a visible ReaderSidebarLists component and expected the Add Button to be hidden

code:
```javascript
{ this.props.subscribedLists && this.props.subscribedLists.length
				? <ReaderSidebarLists
						lists={ this.props.subscribedLists }
						path={ this.props.path }
						isOpen={ this.props.isListsOpen }
						onClick={ this.props.toggleListsVisibility }
						currentListOwner={ this.state.currentListOwner }
						currentListSlug={ this.state.currentListSlug }
					/>
				: null
				}
				<ReaderSidebarLists
						lists={ this.props.subscribedLists }
						path={ this.props.path }
						isOpen={ this.props.isListsOpen }
						onClick={ this.props.toggleListsVisibility }
						currentListOwner={ this.state.currentListOwner }
						currentListSlug={ this.state.currentListSlug }
					/>
				<ReaderSidebarTags
					tags={ this.state.tags }
					path={ this.props.path }
					isOpen={ this.props.isTagsOpen }
					onClick={ this.props.toggleTagsVisibility }
					onTagExists={ this.highlightNewTag }
					currentTag={ this.state.currentTag } />
```

resultr: 


testinglists

**Comment by blowery** [comment]
Nice! That works.  

Looks like we'll need to clean up the fade we use for long list titles a bit though?
comics_ _reader_ _wordpress_com

**Comment by blowery** [comment]
I think you can just drop the `'sidebar__menu-item has-buttons': true,` statement in `reader-sidebar-litsts/list-item.jsx`

**Comment by blowery** [comment]
Moved to #9515

**Pull request closed without merging**